pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1310,6 +1311,20 @@ func (n Nodes) GetResourcesDuration() ResourcesDuration {
 	return i
 }
 
+// GetResourcesDurationByNodeType aggregates resources duration by the given
+// type: DAG, StepGroup, etc.
+func (n Nodes) GetResourcesDurationByNodeType(nodeType NodeType) ResourcesDuration {
+	i := ResourcesDuration{}
+	for _, status := range n {
+		if status.Type != nodeType {
+			continue
+		}
+
+		i = i.Add(status.ResourcesDuration)
```

Comment:
this is only set for Pod nodes - at some point we should sum the pods in a DAG or step to the

Reply:
Gotcha. I'll figure out where the ResourceDuration is being appended to the NodeStatus for pods and do it for the other types too

Reply:
Scratch my previous comment.

@alexec It looks like it would make sense to add a ResourcesDuration property to the Template type, get all of the pods associated with a given Template after it has been executed, and then sum the `DurationForPod` for each pod associated with the given template. Am I going about that correctly?

Reply:
sounds about right

Reply:
Awesome. Will update